hscontrol/derp/server/derp_server.go

```diff
@@ -39,12 +40,22 @@ type DERPServer struct {
 
 func NewDERPServer(
 	serverURL string,
+	verifyHandler http.HandlerFunc,
 	derpKey key.NodePrivate,
 	cfg *types.DERPConfig,
 ) (*DERPServer, error) {
 	log.Trace().Caller().Msg("Creating new embedded DERP server")
 	server := derp.NewServer(derpKey, util.TSLogfWrapper()) // nolint // zerolinter complains
 
+	if cfg.ServerVerifyClients {
+		t := http.DefaultTransport.(*http.Transport)
+		t.RegisterProtocol("headscale", &HeadscaleTransport{
+			verifyHandler: verifyHandler,
+		})
```

Comment:
I am warming up to this quite a bit, but let's make some structural changes, but lets use a bit more of a unique name: `headscale-derp-verify` as the protocol for example.

Maybe this can be done one level further up too? when we initiate headscale so we dont have to pass anything into the server function?